src/screens/documentDetails/components/assignResponsibleModal/index.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useContext } from "react"
+import { InteractiveComponentsContext } from "contexts/interactiveComponentsContext"
+import { Modal } from "components/modal";
+import {
+  Box,
+  FormControl,
+  IconButton,
+  InputLabel,
+  MenuItem,
+  Select,
+  SelectChangeEvent,
+  styled,
+  TextField,
+  Typography,
+} from "@mui/material";
+import CloseIcon from '@mui/icons-material/Close';
+import { LocalizationProvider, DatePicker } from "@mui/x-date-pickers";
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { UserDto } from "services/api/dtos";
+
+interface AssignResponsibleModalProps {
+  assignableResponsibles: UserDto[];
+  responsible: UserDto | undefined;
+  assignResponsible: (userId: string) => void;
+}
+
+export const AssignResponsibleModal = (props: AssignResponsibleModalProps) => {
+  const {
+    isAssignResponsibleModalOpened: isModalOpened,
+    closeAssignResponsibleModal: closeModal,
+  } = useContext(InteractiveComponentsContext);
+
+
+  const {
+    assignableResponsibles,
+    responsible,
+    assignResponsible
+  } = props;
+
+  const onReponsibleChange = (event: SelectChangeEvent) => {
+    assignResponsible(event.target.value as string)
+  }
+
+  return (
+    <Modal
+      isModalOpened={isModalOpened}
+      closeModal={closeModal}
+    >
+      <StyledModalCloseButton
+        aria-label="close"
+        onClick={closeModal}
+      >
+        <CloseIcon />
+      </StyledModalCloseButton>
+      <StyledModalContainer>
+        <form>
```

Comment:
i'd suggest using react hook form to manage the state(values, errors) of the form